src/app/pages/movies-page/movies-page.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { MoviesPageApiService } from './movies-page-api.service';
+import { BehaviorSubject, switchMap, tap } from 'rxjs';
+import { UICardsGridType } from '../../types/ui-types/ui-cards-grid-type';
+
+@Component({
+  selector: 'mdb-movies-page',
+  templateUrl: './movies-page.component.html',
+  styleUrls: ['./movies-page.component.scss'],
+})
+export class MoviesPageComponent {
+  constructor(
+    private router: Router,
+    private route: ActivatedRoute,
+    private moviesApiService: MoviesPageApiService
+  ) {}
+
+  public fullpageGrid = UICardsGridType.fullpage;
+
+  public isLoading$ = new BehaviorSubject<boolean>(true);
+
+  public setPage(newPage: number): void {
+    window.scrollTo({ top: 0, behavior: 'smooth' });
```

Comment:
I'm not sure, but it seems, you could use angular router mechanism for that. Please, check is it related to your case

https://stackoverflow.com/a/51915623
https://angular.io/api/router/ExtraOptions#scrollPositionRestoration

Reply:
Added scroll using this router options. But it feel kinda laggy, i think for production i would not recommend to use it, might cause some troubles, especially in nowadays problematic Safari for iOS.

Reply:
Hm, interesting. I will check this problem later. Now let's keep it like that